Unity/Pyx/Assets/Scripts/GazeGestureManager.cs

```diff
@@ -1,71 +1,67 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
 using UnityEngine;
+using UnityEngine.Networking;
 using UnityEngine.XR.WSA.Input;
 
 public class GazeGestureManager : MonoBehaviour
 {
-    public GameObject jokeText;
+    public GameObject jokeCreatedBy;
+    public GameObject jokeDescription;
+    public GameObject jokeTitle;
 
     public static GazeGestureManager Instance { get; private set; }
 
-    // Represents the hologram that is currently being gazed at.
     public GameObject FocusedObject { get; private set; }
 
     public GameObject sphere;
 
-    int jokeCount = -1;
-
-    List<string> jokes = new List<string>
-    {
-        "I used to hate facial hair, \r\nbut then it grew on me!",
-        "Two guys walk into a bar, the third guy ducks.",
-        "What do you call a bear without any teeth?\r\nA gummy bear!"
-    };
+    int jokeIndex = 1;
 
     GestureRecognizer recognizer;
 
-    // Use this for initialization
+    public float waitTime = 1f;
+
+    float timer;
+
+    private Joke noJoke = new Joke {createdBy = "", title = "Waiting for next instruction", description = ""};
+
     void Awake()
     {
         Instance = this;
 
-        // Set up a GestureRecognizer to detect Select gestures.
         recognizer = new GestureRecognizer();
         recognizer.Tapped += (args) =>
         {
-            // Send an OnSelect message to the focused object and its ancestors.
             if (FocusedObject == sphere)
             {
-                ShowNextJoke();
+                jokeIndex++;
+                FetchJoke();
             }
         };
+
         recognizer.StartCapturingGestures();
 
-        ShowNextJoke();
+        ShowJoke(noJoke);
+
+        FetchJoke();
     }
 
-    // Update is called once per frame
     void Update()
     {
-        // Figure out which hologram is focused this frame.
-        GameObject oldFocusObject = FocusedObject;
+        timer += Time.deltaTime;
+        if (timer > waitTime)
+        {
+            FetchJoke();
```

Comment:
If I'm reading this right - we only need to fetch the joke if the joke to fetch is not equal to the previously fetched joke? I would consider adding a test for that in here (or the FetchJoke method itself) so that we're not performing unnecessary API calls.